### PR TITLE
Feature/merge frontend and backend

### PR DESCRIPTION
**Pull request by chrisgrounds**
I've merged Sam's backend Node branch into my frontend React branch. 

The process in this branch is this: 

(1) cd into repo
(2) cd into chatapp
(2.b) Run `npm install`
(3) run `npm run build`
(4) cd into build
(5) run `node backend.js`
(6) load up browser to localhost:3000

### CODE REVIEW

File: chatapp/public/index.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      You can add webfonts, meta tags, or analytics to this file.
+      The build step will place the bundled scripts into the <body> tag.
+
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
+    -->
+  </body>
+  <script src="/socket.io/socket.io.js"></script>
+  <script src="https://code.jquery.com/jquery-1.11.1.js"></script>
+  <script>
```
Comment (chrisgrounds) [inline chatapp/public/index.html:57]: This is the client Node code

File: chatapp/public/index.html [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    var socket = io();
+    $('form').submit(function(){
+        socket.emit('chat message', $('#m').val());
+        $('#m').val('');
+        return false;
+    });
+    socket.on('chat message', (msg) => {
+        console.log("messaage:" + msg);
+        $('#messages').append($('<li>').text(msg));
```
Comment (chrisgrounds) [inline chatapp/public/index.html:66]: #messages doesn't exist yet, so this doesn't yet work.

File: chatapp/src/App.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.MainHeader {
+    position: absolute;
+    left: 50%;
+    top: 5%;
+    transform: translate(-50%, -50%);
+}
+
+.BelowTitle {
```
Comment (chrisgrounds) [inline chatapp/src/App.css:8]: Beautiful CSS naming conventions being followed here.

File: chatapp/src/registerServiceWorker.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
+// In production, we register a service worker to serve assets from local cache.
```
Comment (chrisgrounds) [inline chatapp/src/registerServiceWorker.js:1]: This was auto-generated.

File: chatapp/public/index.html (hunk 2 above)
Comment (sjoedwards) [reply chatapp/public/index.html:66]: Note ‘messaage’

**Pull request merged**